components/SpotCard.js

```diff
@@ -1,15 +1,57 @@
 import styled from "styled-components";
 import Image from "next/image";
 import placeholderSpotImage from "../assets/images/card-image-placeholder.jpg";
+import { Button } from "./Button";
+import SpotDetails from "./SpotDetails";
+import { useState } from "react";
+
+export default function SpotCard({ spot }) {
+  const [state, setState] = useState("hide");
 
-export default function SpotCard({spot}) {
   return (
     <Card>
       <SpotImage src={placeholderSpotImage} alt="Placeholder Spot Image" />
       <SpotInfos>
         <SpotCategory>{spot.category}</SpotCategory>
         <SpotName>"{spot.name}"</SpotName>
+        <Button
+          type="button"
+          variant="details"
+          onClick={() => {
+            setState(() => {
+              if (state === "hide") {
+                return "show";
+              } else {
+                return "hide";
+              }
+            });
+          }}
+        >
+          Details
+        </Button>
       </SpotInfos>
+      {state === "show" ? (
+        <>
+          <SpotDetails spot={spot} state={state} key={spot.name}></SpotDetails>
+          <Button
+            type="button"
+            variant="cancel"
+            onClick={() => {
+              setState(() => {
+                if (state === "hide") {
+                  return "show";
+                } else {
+                  return "hide";
+                }
+              });
+            }}
+          >
+            Schließen
+          </Button>
+        </>
+      ) : (
+        ""
```

Comment:
`null` is the conventional value for JSX elements when you don't intend on displaying anything.